### PR TITLE
AMR fixes

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Fixes an exploit to let you lase targets outside of zoom.

Fixes lase overlays getting stuck in some cases (Unsure if this is the only cause, I suspect it isn't).
## Why It's Good For The Game
Fixes good.
## Changelog
:cl:
fix: fixed an exploit with the AMR
fix: fixed AMR lase overlays getting stuck sometimes
/:cl:

### CODE REVIEW

File: code/modules/projectiles/guns/specialist.dm [hunk 1]
```diff
@@ -211,13 +212,14 @@ Note that this means that snipers will have a slowdown of 3, due to the scope
...
-		if(laser_target)
-			deactivate_laser_target()
+	SIGNAL_HANDLER
+	if(laser_target)
+		deactivate_laser_target()
 		accuracy_mult -= 0.50 //We lose a big accuracy bonus vs the now unlasered target
 		STOP_PROCESSING(SSobj, src)
 		targetmarker_on = FALSE
 	targetmarker_primed = FALSE
+	UnregisterSignal(src, COMSIG_ITEM_UNZOOM)
```
Comment (TiviPlus) [inline code/modules/projectiles/guns/specialist.dm:222]: You're unregistering src from src anf registering from src to user?

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/specialist.dm (hunk 1 above)
Comment (Lumipharon) [reply code/modules/projectiles/guns/specialist.dm:222]: I am smrt

File: code/modules/projectiles/guns/specialist.dm [hunk 2]
```diff
@@ -203,6 +203,7 @@ Note that this means that snipers will have a slowdown of 3, due to the scope
 		to_chat(user, span_warning("You must be zoomed in to use your target marker!"))
 		return TRUE
 	targetmarker_primed = TRUE //We prime the target laser
+	RegisterSignal(user, COMSIG_ITEM_UNZOOM, .proc/laser_off, user)
```
Comment (TiviPlus) [inline code/modules/projectiles/guns/specialist.dm:206]: second user arg whut

Comment (TiviPlus) [reply code/modules/projectiles/guns/specialist.dm:206]:
```suggestion
	RegisterSignal(user, COMSIG_ITEM_UNZOOM, .proc/laser_off)
```

**Review by TiviPlus — CHANGES_REQUESTED** [review]
rest lgtm i guess

**Pull request merged**